src/videojs-hls.js

```diff
@@ -540,6 +542,11 @@ videojs.HlsHandler.prototype.setCurrentTime = function(currentTime) {
     return currentTime;
   }
 
+  // if we are in the middle of appending a segment, let it finish up
+  if (this.pendingSegment_ && this.pendingSegment_.buffered) {
+    return currentTime;
```

Comment:
Yuck. We need to clean up this workflow but I'm ok with this as a temporary solution.